superset-frontend/src/explore/components/controls/ControlPopover/ControlPopover.tsx

```diff
@@ -44,9 +44,14 @@ export type PopoverProps = BasePopoverProps & {
 const ControlPopover: React.FC<PopoverProps> = ({
   getPopupContainer,
   getVisibilityRatio = getElementYVisibilityRatioOnContainer,
+  visible: visibleProp,
   ...props
 }) => {
   const triggerElementRef = useRef<HTMLElement>();
+
+  const [visible, setVisible] = useState(
```

Comment:
Totally, I don't think that introducing `visible` state here is a good idea. 
We can control `visibleProps`. 
Currently you are mixing `visible` state and `visible` props to control visible, but it is not a good idea.

Reply:
We need the visible state locally to control the popover in this component